spring-rabbit-junit/src/main/java/org/springframework/amqp/rabbit/junit/BrokerRunningSupport.java

```diff
@@ -0,0 +1,655 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeoutException;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.junit.Assume;
+
+import org.springframework.util.Base64Utils;
+import org.springframework.util.StringUtils;
+
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.ConnectionFactory;
+import com.rabbitmq.http.client.Client;
+
+/**
+ * A rule that prevents integration tests from failing if the Rabbit broker application is
+ * not running or not accessible. If the Rabbit broker is not running in the background
+ * all the tests here will simply be skipped (by default) because of a violated assumption
+ * (showing as successful). Usage:
+ *
+ * <pre class="code">
+ * &#064;Rule
+ * public static BrokerRunning brokerIsRunning = BrokerRunning.isRunning();
+ *
+ * &#064;Test
+ * public void testSendAndReceive() throws Exception {
+ * 	// ... test using RabbitTemplate etc.
+ * }
+ * </pre>
+ *
+ * The rule can be declared as static so that it only has to check once for all tests in
+ * the enclosing test case, but there isn't a lot of overhead in making it non-static.
+ * <p>Use {@link #isRunningWithEmptyQueues(String...)} to declare and/or purge test queue(s)
+ * when the rule is run.
+ * <p>Call {@link #removeTestQueues(String...)} from an {@code @After} method to remove
+ * those queues (and optionally others).
+ * <p>If you wish to enforce the broker being available, for example, on a CI server,
+ * set the environment variable {@value #BROKER_REQUIRED} to {@code true} and the
+ * tests will fail fast.
+ *
+ * @author Dave Syer
+ * @author Gary Russell
+ *
+ * @since 1.7
+ * @see Assume
+ * @see org.junit.internal.AssumptionViolatedException
+ */
+public final class BrokerRunningSupport {
+
+	private static final int SIXTEEN = 16;
+
+	public static final String BROKER_ADMIN_URI = "RABBITMQ_TEST_ADMIN_URI";
+
+	public static final String BROKER_HOSTNAME = "RABBITMQ_TEST_HOSTNAME";
+
+	public static final String BROKER_PORT = "RABBITMQ_TEST_PORT";
+
+	public static final String BROKER_USER = "RABBITMQ_TEST_USER";
+
+	public static final String BROKER_PW = "RABBITMQ_TEST_PASSWORD";
+
+	public static final String BROKER_ADMIN_USER = "RABBITMQ_TEST_ADMIN_USER";
+
+	public static final String BROKER_ADMIN_PW = "RABBITMQ_TEST_ADMIN_PASSWORD";
+
+	public static final String BROKER_REQUIRED = "RABBITMQ_SERVER_REQUIRED";
+
+	public static final String DEFAULT_QUEUE_NAME = BrokerRunningSupport.class.getName();
+
+	private static final String GUEST = "guest";
+
+	private static final Log logger = LogFactory.getLog(BrokerRunningSupport.class); // NOSONAR - lower case
+
+	// Static so that we only test once on failure: speeds up test suite
+	private static final Map<Integer, Boolean> brokerOnline = new HashMap<Integer, Boolean>(); // NOSONAR - lower case
```

Comment:
I guess we can fix all these Sonar smells using upper case already. Plus we can use now diamonds.